test/gurka/test_expansion.cc

```diff
@@ -0,0 +1,181 @@
+#include "gurka.h"
+#include <boost/algorithm/string/join.hpp>
+#include <boost/format.hpp>
+#include <gtest/gtest.h>
+
+using namespace valhalla;
+
+class ExpansionTest : public ::testing::TestWithParam<std::vector<std::string>> {
+protected:
+  static gurka::map expansion_map;
+
+  static void SetUpTestSuite() {
+    constexpr double gridsize = 10;
+
+    // 5 birectional edges, 1 oneway = 11 directed edges
+    const std::string ascii_map = R"(
+            B  F--G
+            |  |  |
+         E--A--C--H
+            |
+            D
+    )";
+
+    const gurka::ways ways = {
+        {"DAB", {{"highway", "primary"}}},
+        {"EACH", {{"highway", "primary"}}},
+        {"CFG", {{"highway", "primary"}, {"oneway", "yes"}}},
+    };
+
+    const auto layout = gurka::detail::map_to_coordinates(ascii_map, gridsize);
+    expansion_map = gurka::buildtiles(layout, ways, {}, {}, "test/data/expansion");
+  }
+
+  // common method can't deal with arrays of strings
+  std::string build_local_req(rapidjson::Document& doc,
+                              rapidjson::MemoryPoolAllocator<>& allocator,
+                              const std::vector<std::string>& waypoints,
+                              const std::string& costing,
+                              const std::string& action,
+                              bool skip_opps,
+                              const std::vector<std::string> props) {
```

Comment:
see how to remove: https://github.com/gis-ops/valhalla/blob/b37eb477b5d024868d31dfb6612fbe47bb88fad2/test/gurka/test_route.cc#L403